modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DefaultEventHandlersPropsTransformer.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export default function propsTransformer({
+	defaultEventHandler,
+	items,
+	...otherProps
+}) {
+	return {
+		...otherProps,
+		items: items.map((item) => {
+			return {
+				...item,
+				onClick(event) {
+					const action = item.data?.action;
+
+					if (action) {
+						event.preventDefault();
+
+						Liferay.componentReady(defaultEventHandler).then(
+							(eventHandler) => {
+								eventHandler[action](item.data);
+							}
+						);
+					}
+				},
+			};
+		}),
+	};
+}
```

Comment:
Something about this reminds me of...

![star-destroyer](https://user-images.githubusercontent.com/7074/86264406-c4525100-bbc2-11ea-9d89-c25022830f7c.jpg)

Reply:
Any suggestions to defeat the evil empire? 😂

Reply:
Who said this was a problem?

https://youtu.be/U1htC8Rf3qQ

Reply:
![droids](https://user-images.githubusercontent.com/905006/86366225-d099e500-bc7a-11ea-8635-24f31c2019f3.gif)